core/src/main/scala/org/apache/spark/util/logging/FileAppender.scala

```diff
@@ -76,7 +76,11 @@ private[spark] class FileAppender(inputStream: InputStream, file: File, bufferSi
           }
         }
       } {
-        closeFile()
+        try {
+          inputStream.close()
```

Comment:
Yes,you are right.
But this function is only used in `ExecutorRunner`,  also if an exception occurs within this function,this will ensure the inputStream  is closed